### ISSUE TITLE
Docs on global var produce different outputs

### ISSUE THREAD

**Issue by jm-positron** [astral-sh/ruff#18139]
### Summary

The docs on this page are inaccurate: https://docs.astral.sh/ruff/rules/global-statement/

The two examples produce slightly different outputs.

The original example produces:

```
10
10
```
vs
```
1
10
```

The examples should produce identical output.

Note I spent some time trying to find where in the project those docs are produced, but unfortuntately wasn't able to find it quickly. So this report just references the url.

**Comment by ntBre** [comment]
Yeah, it can be a bit tricky to map the rules back to their source code. The docs are extracted from the docstring [here](https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/rules/pylint/rules/global_statement.rs#L6) if you're interested in updating it :)

I'll work on it or add the `help-wanted` label if not.

**Comment by jm-positron** [comment]
I think it would be better for someone more familiar with ruff and its objectives to update it. Thank you for the link to the doc, I believe I failed to find it when searching for a string from the docs due to the markdown code backticks.
